### DISCUSSION TITLE
Poetry Fails to Resolve Transitive Dependencies from AWS CodeArtifact

### DISCUSSION THREAD

**Discussion by ArKhachatryan** [python-poetry/poetry#8936] [Q&A]
I am encountering an issue where Poetry successfully resolves and installs a package (package-A) from an AWS CodeArtifact repository but fails to resolve its transitive dependencies (such as package-B), which are also hosted in the same CodeArtifact repository. Both projects (package-A and main-project) are configured to use this repository, and I have set the necessary environment variables to interact with AWS CodeArtifact.

**pyproject.toml of main-project**

[tool.poetry]
name = "main-project"
version = "0.1.0"
description = "desc"
authors = ["am"]

[[tool.poetry.source]]
name = "aws"
url = "https://my-codeartifact-url.d.codeartifact.region.amazonaws.com/pypi/my-repo/simple"
priority="explicit"

[tool.poetry.dependencies]
python = "3.11.7"
package-A = {version = "0.3.5", source = "aws"}

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"

**pyproject.toml of package-A**

[tool.poetry]
name = "package-A"
version = "0.3.5"
description = "desc"
authors = ["am"]

[[tool.poetry.source]]
name = "aws"
url = "https://my-codeartifact-url.d.codeartifact.region.amazonaws.com/pypi/my-repo/simple"
priority="explicit"

[tool.poetry.dependencies]
python = "3.11.7"
package-B = {version = "0.5.5", source = "aws"}

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"

**Expected Behavior**
Poetry should resolve and install package-A and its dependencies from the AWS CodeArtifact repository without any issues, as the repository URL and authentication are correctly configured.

**Current Behavior**
While Poetry successfully installs package-A from the CodeArtifact repository, it fails to find and install its dependencies (package-B), which are also available in the same repository. The installation process errors out, indicating that it cannot find the package-B package.

**Environment**
OS: Ubuntu 22.04
Poetry Version: 1.7.1
Python Version: 3.11.7

**Comment by abn** [comment]
Does B appear in your lockfile?

Can you check the METADATA in the published wheel of A?
